diff --git a/datatables.module b/datatables.module
index 79cdb6e..193c892 100644
--- a/datatables.module
+++ b/datatables.module
@@ -305,7 +305,7 @@ function template_preprocess_datatables_view(&$vars) {
     // Set default sort order.
     $datatable_options['aaSorting'] = array();
     if ($options['default'] == $field) {
-      $datatable_options['aaSorting'] = array(array($position, $options['order']));
+      $datatable_options['aaSorting'] = array(array($position, $options['info'][$field]['default_sort_order']));
     }
 
     $position++;
